src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -3055,17 +3081,17 @@ public override BoundNode VisitDynamicIndexerAccess(BoundDynamicIndexerAccess no
         private void CheckPossibleNullReceiver(BoundExpression receiverOpt, bool checkType = true)
         {
             if (receiverOpt != null &&
-                (!checkType || ((object)receiverOpt.Type != null && receiverOpt.Type.IsReferenceType)) &&
+                (!checkType ||  ((object)receiverOpt.Type != null && (receiverOpt.Type.IsReferenceType || receiverOpt.Type.IsUnconstrainedTypeParameter()))) &&
                 this.State.Reachable &&
-                _result.Type?.IsNullable == true)
+                (_result.Type?.IsNullable == true || IsUnconstrainedTypeParameter(_result.Type?.TypeSymbol)))
```

Comment:
>_result.Type?.TypeSymbol [](start = 82, length = 24)

It feels strange that we are doing this check for different type symbols within the same ```if```, when it appears we must be talking about the same type.

Reply:
> It feels strange that we are doing this check for different type symbols within the same if, when it appears we must be talking about the same type.

This comment is still marked as active and it doesn't look like there was any response for it.

---
In reply to: [177554170](https://github.com/dotnet/roslyn/pull/25733#discussion_r177554170) [](ancestors = 177554170)